test/runtests.jl

```diff
@@ -165,17 +165,17 @@ end
     @test ϕuP(1,Ip,I,f,1)==λ(f[Ip],f[I-δ(1,I)],f[I])
 
     # check applying acceleration
-    N = 4
-    a = zeros(N,N,2)
-    WaterLily.accelerate!(a,1,nothing,())
-    @test all(a .== 0)
-    WaterLily.accelerate!(a,1,(i,t) -> i==1 ? t : 2*t,())
-    @test all(a[:,:,1] .== 1) && all(a[:,:,2] .== 2)
-    WaterLily.accelerate!(a,1,nothing,(i,t) -> i==1 ? -t : -2*t)
-    @test all(a[:,:,1] .== 0) && all(a[:,:,2] .== 0)
-    WaterLily.accelerate!(a,1,(i,t) -> i==1 ? t : 2*t,(i,t) -> i==1 ? -t : -2*t)
-    @test all(a[:,:,1] .== 0) && all(a[:,:,2] .== 0)
-
+    for f ∈ arrays
+        N = 4; a = zeros(N,N,2) |> f
+        WaterLily.accelerate!(a,1,nothing,())
+        @test all(a .== 0)
```

Comment:
missing the `allowscalar` for all these tests?

Reply:
I double-checked, and the GPU only complains if I do `a[1,1,1]==0`, he is happy in this case :)